File: invision-backend/app/Enums/VisitStatus.php
<?php

namespace App\Enums;

enum VisitStatus: string
{
    case Pending = 'pending';
    case CheckedIn = 'checked_in';
    case Completed = 'completed';
    case Skipped = 'skipped';

    public function label(): string
    {
        return match ($this) {
            self::Pending => 'Pending',
            self::CheckedIn => 'Checked In',
            self::Completed => 'Completed',
            self::Skipped => 'Skipped',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Pending => 'gray',
            self::CheckedIn => 'blue',
            self::Completed => 'green',
            self::Skipped => 'red',
        };
    }
}

File: invision-mobile/lib/src/core/enums/visit_status.dart
enum VisitStatus {
  pending,
  checkedIn,
  completed,
  skipped;

  String get label => switch (this) {
    pending => 'Pending',
    checkedIn => 'Checked In',
    completed => 'Completed',
    skipped => 'Skipped',
  };

  static VisitStatus fromString(String value) {
    return switch (value) {
      'pending' => VisitStatus.pending,
      'checked_in' => VisitStatus.checkedIn,
      'completed' => VisitStatus.completed,
      'skipped' => VisitStatus.skipped,
      _ => VisitStatus.pending,
    };
  }
}
